src/analysermodel.cpp

```diff
@@ -348,4 +349,21 @@ bool AnalyserModel::needAcothFunction() const
     return mPimpl->mNeedAcothFunction;
 }
 
+bool AnalyserModel::areEquivalentVariables(const VariablePtr &variable1,
+                                           const VariablePtr &variable2)
+{
+    auto key = reinterpret_cast<intptr_t>(variable1.get()) * reinterpret_cast<intptr_t>(variable2.get());
+    auto cacheKey = mPimpl->mCachedEquivalentVariables.find(key);
+
+    if (cacheKey != mPimpl->mCachedEquivalentVariables.end()) {
+        return cacheKey->second;
+    }
+
+    bool res = libcellml::areEquivalentVariables(variable1, variable2);
+
+    mPimpl->mCachedEquivalentVariables[key] = res;
+
+    return res;
+}
```

Comment:
Another question: This is the only function which affects the cache?  If so, where is the cache cleared so that new/old/changed variable equivalences are updated?  I realise that the Analyser assumes that the model it's been given is static, but what happens if the Analyser is called twice? or used to process two different models? or one model with a change in variable equivalence?  I have a feeling I'm missing something here :(

Reply:
When you analyse a model (see `Analyser::AnalyserImpl::analyseModel()`) we create an instance of `AnalyserModel`, which is then populated with whatever we want the user to access from such an instance. As part of that process, the cache gets automatically built (through calls to `AnalyserModel::areEquivalentVariables()`). This is something that happens only once, so no need for clearing the cache or anything.

Reply:
OK, that makes (a little more) sense ... so the caching is **only** to speed up repeated calls to test equivalence between the same two variables?  And those calls are only expected to happen internally to the `Analyser` class?

Reply:
That is all correct. Then, when it comes to `AnalyserModel::areEquivalentVariables()`, we make it available through our API because we never know: an end-user might want to take advantage of it.

Reply:
OK, thanks.  In that case I think I second what @nickerso said about the documentation for this function.  It needs to be really clear that calls to `areEquivalentVariables` on the `AnalyserModel` are fundamentally different from similar calls on the `Variable` and `Model` classes.

Reply:
Done in commit 87eb46f6b.